mlos_core/mlos_core/optimizers/bayesian_optimizers.py

```diff
@@ -276,6 +277,177 @@ def surrogate_predict(self, configurations: pd.DataFrame, context: Optional[pd.D
+        from smac.intensifier.abstract_intensifier import AbstractIntensifier # pylint: disable=import-outside-toplevel
+        from smac.initial_design import LatinHypercubeInitialDesign # pylint: disable=import-outside-toplevel
+        from smac.main.config_selector import ConfigSelector # pylint: disable=import-outside-toplevel
+        from smac.random_design.probability_design import ProbabilityRandomDesign # pylint: disable=import-outside-toplevel
+        from smac.runhistory import TrialInfo # pylint: disable=import-outside-toplevel
+
+        self.seed: int = seed or 0
+        self.trial_info_map: Mapping[ConfigSpace.Configuration, TrialInfo] = {} # Stores TrialInfo instances returned by .ask()
+
+        # Instantiate Scenario
+        output_directory = Path(output_directory)
+        scenario: Scenario = Scenario(
+            self.optimizer_parameter_space,
+            name=run_name,
+            output_directory=output_directory,
+            deterministic=True,
+            n_trials=1e4,
+            seed=self.seed,
+            n_workers=n_workers,
+        )
+        intensifier: AbstractIntensifier = Optimizer_Smac.get_intensifier(scenario, max_config_calls=1)
+        config_selector: ConfigSelector = ConfigSelector(scenario, retrain_after=1)
+
+        # Customize SMAC's randomized behavior
+        initial_design: Optional[LatinHypercubeInitialDesign] = None
+        if n_random_init is not None:
+            initial_design = LatinHypercubeInitialDesign(scenario=scenario, n_configs=n_random_init)
+        random_design: Optional[ProbabilityRandomDesign] = None
+        if n_random_probability is not None:
+            random_design = ProbabilityRandomDesign(probability=n_random_probability)
+
+        # Create SMAC optimizer
+        self.base_optimizer = Optimizer_Smac(
+            scenario,
+            SmacOptimizer._dummy_target_func,
+            initial_design=initial_design,
+            intensifier=intensifier,
+            random_design=random_design,
+            config_selector=config_selector,
+            overwrite=True,
+        )
+
+    @staticmethod
+    def _dummy_target_func(seed):
+        """Dummy target function for SMAC optimizer.
+
+        Since we only use the ask-and-tell interface, this is never called.
+        """
+        # NOTE: Providing a target function when using the ask-and-tell interface is an imperfection of the API
+        # -- this planned to be fixed in some future release: https://github.com/automl/SMAC3/issues/946
+        raise RuntimeError('This function should never be called.')
+
+    def _register(self, configurations: pd.DataFrame, scores: pd.Series, context: Optional[pd.DataFrame] = None) -> None:
+        """Registers the given configurations and scores.
+
+        Parameters
+        ----------
+        configurations : pd.DataFrame
+            Dataframe of configurations / parameters. The columns are parameter names and the rows are the configurations.
+
```

Comment:
remove blank lines between the parameters in docstrings. they make flake8 very unhappy.
(also, maybe, run `flake8` in addition to `mypy` and `pylint --disable="fixme,consider-alternative-union-syntax"` for the entire project?)